### PR TITLE
Bugfix 258 harmonize local http server response with vertx behaviour

### PR DESCRIPTION
**Pull request by oli-h**
Solves #258 
See commit comments for details

### CODE REVIEW

File: gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java [hunk 1]
```diff
@@ -395,21 +380,29 @@ private HttpClientRequest prepareRequest(final HttpServerRequest req, final Stri
...
                 error("Problem with backend: " + exception.getMessage(), req, targetUri);
-                req.response().setStatusCode(StatusCode.INTERNAL_SERVER_ERROR.getStatusCode());
-                req.response().setStatusMessage(StatusCode.INTERNAL_SERVER_ERROR.getStatusMessage());
-                try {
-                    ResponseStatusCodeLogUtil.info(req, StatusCode.INTERNAL_SERVER_ERROR, Forwarder.class);
-                    req.response().end(req.response().getStatusMessage());
-                } catch (IllegalStateException e) {
-                    // ignore because maybe already closed
-                }
+                respondError(req, StatusCode.INTERNAL_SERVER_ERROR);
```
[the remark this answers is not in the document]
Comment (oli-h) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java]: Thx, nonvisiblebeta. Could not add you as a reviewer in advance

File: gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java [hunk 2]
```diff
@@ -395,21 +380,29 @@ private HttpClientRequest prepareRequest(final HttpServerRequest req, final Stri
...
+            ResponseStatusCodeLogUtil.info(req, statusCode, Forwarder.class);
+
+            String msg = statusCode.getStatusMessage();
+            req.response()
+                    .setStatusCode(statusCode.getStatusCode())
+                    .setStatusMessage(msg)
+                    .end(msg);
+        } catch (IllegalStateException ex) {
+            // (nearly) ignore because underlying connection maybe already closed
+            RequestLoggerFactory.getLogger(Forwarder.class, req).info("IllegalStateException while sending error response for {}", req.uri(), ex);
```
Comment (hiddenalpha) [inline gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java]: Is it correct to use _RequestLogger_ here?

Comment (oli-h) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java]: Not 100% sure - I don't completly understand this Logging-Infrastructure

Comment (mcweba) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java]: As you figured out correctly, the RequestLoggerFactory adds the request-id (when available) to the log statement to keep the link between log entry and request. You can refactor the RequestLoggerFactory as well if you want :-)
I agree to use this "log construct" here.

**Pull request merged**